src/storage/txn/process.rs

```diff
@@ -707,8 +713,21 @@ fn process_write_impl<S: Snapshot>(
                         .as_mut()
                         .unwrap()
                         .entry((current_lock.ts, current_lock.for_update_ts != 0))
-                        .or_insert(vec![])
-                        .push(lock_manager::gen_key_hash(&current_key));
+                        .and_modify(|key_hashes: &mut Option<Vec<u64>>| {
+                            if key_hashes.is_some() {
+                                key_hashes
+                                    .as_mut()
+                                    .unwrap()
+                                    .push(lock_manager::gen_key_hash(&current_key));
+                            }
+                        })
+                        .or_insert_with(|| {
+                            if wait_table_is_empty {
+                                None
+                            } else {
+                                Some(vec![])
```

Comment:
I think we should insert current keys into the empty vec here.

Reply:
Nice catch!